### PR TITLE
chore: tests for client initialize

### PR DESCRIPTION
**Pull request by suthar26**
client initialization tests working

### CODE REVIEW

**Comment by suthar26** [comment]
WIP

File: harness/features/initialize.cloud.test.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            const res = await fetch(`${url}/spec`)
+            const response = await res.json()
+
+            expect(response.name).toBeDefined()
+            expect(response.capabilities).toBeDefined()
+        })
+
+        describeIf(capabilities.includes(Capabilities.cloud))(name, () => {
+
+            it.only('should throw an exceptinon and return no location if invalid SDK token is sent', async () => {
```
Comment (emmawillis) [inline harness/features/initialize.cloud.test.ts:35]: should be able to remove the `.only` on these

File: harness/features/initialize.cloud.test.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            it('should return client object location if SDK token is correct', async () => {
+                const clientId: string = uuidv4()
+                const response = await createClient(url, clientId, sdkKey,
+                    { ...clientOptions, baseURLOverride: `${mockServerUrl}/client/${clientId}` })
+
+                scope
+                    .get(`/client/${clientId}/config/v1/server/${sdkKey}.json`)
+                    .reply(200, {})
+
+                await wait(1000)
```
Comment (elliotCamblor) [inline harness/features/initialize.cloud.test.ts:62]: I dont think you should need this `wait` here, the cloudClient initialization process has nothing asynchronous that you need to wait for

File: harness/features/initialize.cloud.test.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+                const body = await response.json()
+                expect(body.message).toBe('success')
+                const createdClientId = response.headers.get('location')
+                expect(createdClientId).toBe(`client/${clientId}`)
+                expect(scope.isDone).toBeTruthy()
```
Comment (elliotCamblor) [inline harness/features/initialize.cloud.test.ts:68]: `scope.isDone()` should be a function call

File: harness/features/initialize.cloud.test.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+                scope
+                    .get(`/client/${clientId}/config/v1/server/${sdkKey}.json`)
+                    .reply(200, {})
+
+                await wait(1000)
+
+                const body = await response.json()
+                expect(body.message).toBe('success')
+                const createdClientId = response.headers.get('location')
```
[the remark this answers is not in the document]
Comment (suthar26) [reply harness/features/initialize.cloud.test.ts:66]: specifically for this , we get confirmation that the initialize call was successful on the client?

[the remark this answers is not in the document]
Comment (ajwootto) [reply harness/features/initialize.cloud.test.ts:66]: don't we already know that by the lack of "exception" field in the response or something else?

[the remark this answers is not in the document]
Comment (suthar26) [reply harness/features/initialize.cloud.test.ts:66]: ah yes already checking for the `body.message` to be `success`

**Review by elliotCamblor — APPROVED** [review]

**Pull request merged**